coord/src/rmr/rmr.c

```diff
@@ -500,7 +508,9 @@ int MR_UpdateTopology(MRClusterTopology *newTopo) {
   rc->ctx = newTopo;
   rc->cb = uvUpdateTopologyRequest;
   rc->protocol = 0;
-  RQ_Push(rq_g, requestCb, rc);
+  /* This request is called periodically and might be still in the queue
+  during a shut down event. see RQ_Push comment*/
+  RQ_Push(rq_g, requestCb, rc, freeUpdateTopologyRequest);
```

Comment:
In general I think it makes more sense to add a free callback to `MRRequestCtx` and then have a general free function that frees `MRRequestCtx` and calls this free function on the `rc->ctx`. But because we are doing all this just to avoid the errors on our sanitizer I would not insist on it.

Reply:
that means that we are holding a free callback in the queueItem + the request (because the queue can't call functions from the request API directly unless he has a pointer to them)
Isn't it redundant?

Reply:
I think that for feature completeness and responsibility isolation its not redundant. But also I understand the purpose of why we are doing it and so its not critical.